example/spec/helpers/granuleUtils.js

```diff
@@ -88,45 +90,57 @@ function createGranuleFiles(granuleFiles, bucket, oldGranuleId, newGranuleId) {
  * @returns {Promise<Object>} - input payload as a JS object with the updated granule ids
  */
 async function setupTestGranuleForIngest(bucket, inputPayloadJson, granuleRegex, testSuffix = '', testDataFolder = null) {
-  // granule id for the new files
-  const newGranuleId = randomStringFromRegex(granuleRegex);
-  console.log(`\ngranule id: ${newGranuleId}`);
+  const payloadJson = testDataFolder ?
+    globalReplaceC('cumulus-test-data/pdrs', testDataFolder, inputPayloadJson) :
+    inputPayloadJson;
+
+  const baseInputPayload = JSON.parse(payloadJson);
 
-  if (testDataFolder) inputPayloadJson = globalReplace(inputPayloadJson, 'cumulus-test-data/pdrs', testDataFolder); //eslint-disable-line no-param-reassign
-  const baseInputPayload = JSON.parse(inputPayloadJson);
   const oldGranuleId = baseInputPayload.granules[0].granuleId;
   baseInputPayload.granules[0].dataType += testSuffix;
 
+  const newGranuleId = randomStringFromRegex(granuleRegex);
+
   await createGranuleFiles(
     baseInputPayload.granules[0].files,
     bucket,
     oldGranuleId,
     newGranuleId
   );
 
-  const baseInputPayloadJson = JSON.stringify(baseInputPayload);
-  const updatedInputPayloadJson = globalReplace(baseInputPayloadJson, oldGranuleId, newGranuleId);
-
-  return JSON.parse(updatedInputPayloadJson);
+  return thread(
```

Comment:
I like this use of the curried globalReplace function. 👍

Reply:
If you like that, just _wait_ until you see line 137 😄